### ISSUE TITLE
Last commit broke 'matrix_display_switch'

### ISSUE THREAD

**Issue by marianomd** [TillFleisch/ESPHome-HUB75-MatrixDisplayWrapper#10]
Hello and thanks for this wrapper.

The last commit (1675d24) broke the switch functionality.

yaml:
```
switch:
  - platform: matrix_display_switch
    matrix_id: matrix
    name: "Power"
    id: power
    restore_mode: ALWAYS_ON
```

Compile output:
```
INFO ESPHome 2023.11.6
INFO Reading configuration /config/esphome/pixelclock.yaml...
INFO Detected timezone 'America/Argentina/Buenos_Aires'
INFO Generating C++ source...
INFO Compiling app...
Processing pixelclock (board: esp32dev; framework: arduino; platform: platformio/espressif32@5.4.0)
--------------------------------------------------------------------------------
HARDWARE: ESP32 240MHz, 320KB RAM, 4MB Flash
 - toolchain-xtensa-esp32 @ 8.4.0+2021r2-patch5
Dependency Graph
|-- SPI @ 2.0.0
|-- Wire @ 2.0.0
|-- Adafruit BusIO @ 1.14.5
|-- Adafruit GFX Library @ 1.11.9
|-- ESP32 HUB75 LED MATRIX PANEL DMA Display @ 3.0.9+sha.d4525a0
|-- AsyncTCP-esphome @ 2.0.1
|-- WiFi @ 2.0.0
|-- FS @ 2.0.0
|-- Update @ 2.0.0
|-- ESPAsyncWebServer-esphome @ 3.1.0
|-- DNSServer @ 2.0.0
|-- ESPmDNS @ 2.0.0
|-- noise-c @ 0.1.4
|-- ArduinoJson @ 6.18.5
Compiling .pioenvs/pixelclock/src/esphome/components/matrix_display/matrix_display.cpp.o
Compiling .pioenvs/pixelclock/src/esphome/components/matrix_display_brightness/matrix_display_brightness.cpp.o
Compiling .pioenvs/pixelclock/src/esphome/components/matrix_display_switch/matrix_display_switch.cpp.o
In file included from src/esphome/components/matrix_display/matrix_display.cpp:1:
src/esphome/components/matrix_display/matrix_display.h:32:12: error: 'void esphome::matrix_display::MatrixDisplay::setup()' marked 'override', but does not override
       void setup() override;
            ^~~~~
src/esphome/components/matrix_display/matrix_display.h:34:12: error: 'void esphome::matrix_display::MatrixDisplay::dump_config()' marked 'override', but does not override
       void dump_config() override;
            ^~~~~~~~~~~
src/esphome/components/matrix_display/matrix_display.h:36:12: error: 'void esphome::matrix_display::MatrixDisplay::update()' marked 'override', but does not override
       void update() override;
            ^~~~~~
src/esphome/components/matrix_display/matrix_display.cpp: In member function 'void esphome::matrix_display::MatrixDisplay::setup()':
src/esphome/components/matrix_display/matrix_display.cpp:36:42: error: 'update_interval_' was not declared in this scope
       mxconfig.min_refresh_rate = 1000 / update_interval_;
                                          ^~~~~~~~~~~~~~~~
src/esphome/components/matrix_display/matrix_display.cpp:36:42: note: suggested alternative: 'init_internal_'
       mxconfig.min_refresh_rate = 1000 / update_interval_;
                                          ^~~~~~~~~~~~~~~~
                                          init_internal_
In file included from src/esphome/components/matrix_display_brightness/matrix_display_brightness.h:5,
                 from src/esphome/components/matrix_display_brightness/matrix_display_brightness.cpp:1:
src/esphome/components/matrix_display_brightness/../matrix_display/matrix_display.h:32:12: error: 'void esphome::matrix_display::MatrixDisplay::setup()' marked 'override', but does not override
       void setup() override;
            ^~~~~
src/esphome/components/matrix_display_brightness/../matrix_display/matrix_display.h:34:12: error: 'void esphome::matrix_display::MatrixDisplay::dump_config()' marked 'override', but does not override
       void dump_config() override;
            ^~~~~~~~~~~
src/esphome/components/matrix_display_brightness/../matrix_display/matrix_display.h:36:12: error: 'void esphome::matrix_display::MatrixDisplay::update()' marked 'override', but does not override
       void update() override;
            ^~~~~~
In file included from src/esphome/components/matrix_display_switch/matrix_display_switch.h:5,
                 from src/esphome/components/matrix_display_switch/matrix_display_switch.cpp:1:
src/esphome/components/matrix_display_switch/../matrix_display/matrix_display.h:32:12: error: 'void esphome::matrix_display::MatrixDisplay::setup()' marked 'override', but does not override
       void setup() override;
            ^~~~~
src/esphome/components/matrix_display_switch/../matrix_display/matrix_display.h:34:12: error: 'void esphome::matrix_display::MatrixDisplay::dump_config()' marked 'override', but does not override
       void dump_config() override;
            ^~~~~~~~~~~
src/esphome/components/matrix_display_switch/../matrix_display/matrix_display.h:36:12: error: 'void esphome::matrix_display::MatrixDisplay::update()' marked 'override', but does not override
       void update() override;
            ^~~~~~
*** [.pioenvs/pixelclock/src/esphome/components/matrix_display/matrix_display.cpp.o] Error 1
*** [.pioenvs/pixelclock/src/esphome/components/matrix_display_brightness/matrix_display_brightness.cpp.o] Error 1
Compiling .pioenvs/pixelclock/src/main.cpp.o
*** [.pioenvs/pixelclock/src/esphome/components/matrix_display_switch/matrix_display_switch.cpp.o] Error 1
In file included from src/esphome.h:32,
                 from src/main.cpp:3:
src/esphome/components/matrix_display/matrix_display.h:32:12: error: 'void esphome::matrix_display::MatrixDisplay::setup()' marked 'override', but does not override
       void setup() override;
            ^~~~~
src/esphome/components/matrix_display/matrix_display.h:34:12: error: 'void esphome::matrix_display::MatrixDisplay::dump_config()' marked 'override', but does not override
       void dump_config() override;
            ^~~~~~~~~~~
src/esphome/components/matrix_display/matrix_display.h:36:12: error: 'void esphome::matrix_display::MatrixDisplay::update()' marked 'override', but does not override
       void update() override;
            ^~~~~~
src/main.cpp: In function 'void setup()':
src/main.cpp:3015:61: error: no matching function for call to 'esphome::UpdateComponentAction<>::UpdateComponentAction(esphome::matrix_display::MatrixDisplay*&)'
   updatecomponentaction = new UpdateComponentAction<>(matrix);
                                                             ^
In file included from src/esphome.h:71,
                 from src/main.cpp:3:
src/esphome/core/base_automation.h:337:3: note: candidate: 'esphome::UpdateComponentAction<Ts>::UpdateComponentAction(esphome::PollingComponent*) [with Ts = {}]'
   UpdateComponentAction(PollingComponent *component) : component_(component) {}
   ^~~~~~~~~~~~~~~~~~~~~
src/esphome/core/base_automation.h:337:3: note:   no known conversion for argument 1 from 'esphome::matrix_display::MatrixDisplay*' to 'esphome::PollingComponent*'
src/esphome/core/base_automation.h:335:32: note: candidate: 'constexpr esphome::UpdateComponentAction<>::UpdateComponentAction(const esphome::UpdateComponentAction<>&)'
 template<typename... Ts> class UpdateComponentAction : public Action<Ts...> {
                                ^~~~~~~~~~~~~~~~~~~~~
src/esphome/core/base_automation.h:335:32: note:   no known conversion for argument 1 from 'esphome::matrix_display::MatrixDisplay*' to 'const esphome::UpdateComponentAction<>&'
src/esphome/core/base_automation.h:335:32: note: candidate: 'constexpr esphome::UpdateComponentAction<>::UpdateComponentAction(esphome::UpdateComponentAction<>&&)'
src/esphome/core/base_automation.h:335:32: note:   no known conversion for argument 1 from 'esphome::matrix_display::MatrixDisplay*' to 'esphome::UpdateComponentAction<>&&'
*** [.pioenvs/pixelclock/src/main.cpp.o] Error 1
========================== [FAILED] Took 6.62 seconds ==========================
```

Cheers,
Mariano

**Comment by marianomd** [comment]
I just realized it was to support esphome 2023.12. Upgrading esphome solves the issue.

**Comment by TillFleisch** [comment]
#9 fixed build issues related to ESPHome 2023.12, since you are using ESPHome 2023.11.x the latest commit will not compile. Either update to the latest version of ESPHome (2023.12 or higher) or use a commit older than https://github.com/TillFleisch/ESPHome-HUB75-MatrixDisplayWrapper/commit/1675d2434aba34c095de4cdfca57c734cec31238.

**Comment by marianomd** [comment]
Thanks for the prompt response
